clients/include/blas2/testing_spr_batched.hpp

```diff
@@ -166,76 +167,102 @@ void testing_spr_batched(const Arguments& arg)
-    host_batch_vector<T> hA_host(A_size, 1, batch_count);
-    host_batch_vector<T> hA_device(A_size, 1, batch_count);
+    // Naming: `h` is in CPU (host) memory(eg hAp), `d` is in GPU (device) memory (eg dAp).
+    // Allocate host memory
+    host_batch_matrix<T> hA(N, N, N, batch_count);
+    host_batch_matrix<T> hAp(1, size_A, 1, batch_count);
+    host_batch_matrix<T> hAp_host(1, size_A, 1, batch_count);
+    host_batch_matrix<T> hAp_device(1, size_A, 1, batch_count);
+    host_batch_matrix<T> hAp_cpu(1, size_A, 1, batch_count);
     host_batch_vector<T> hx(N, incx, batch_count);
+    host_vector<T>       halpha(1);
 
-    device_batch_vector<T> dA(A_size, 1, batch_count);
+    // Check host memory allocation
+    CHECK_HIP_ERROR(hA.memcheck());
+    CHECK_HIP_ERROR(hAp.memcheck());
+    CHECK_HIP_ERROR(hAp_host.memcheck());
+    CHECK_HIP_ERROR(hAp_device.memcheck());
+    CHECK_HIP_ERROR(hAp_cpu.memcheck());
+    CHECK_HIP_ERROR(hx.memcheck());
+
+    halpha[0] = h_alpha;
+
+    // Allocate device memory
+    device_batch_matrix<T> dAp(1, size_A, 1, batch_count);
     device_batch_vector<T> dx(N, incx, batch_count);
     device_vector<T>       d_alpha(1);
 
-    CHECK_HIP_ERROR(dA.memcheck());
-    CHECK_HIP_ERROR(dx.memcheck());
+    // Check device memory allocation
+    CHECK_DEVICE_ALLOCATION(dAp.memcheck());
+    CHECK_DEVICE_ALLOCATION(dx.memcheck());
+    CHECK_DEVICE_ALLOCATION(d_alpha.memcheck());
 
-    hipblas_init_vector(hA, arg, hipblas_client_never_set_nan, true);
+    // Initial Data on CPU
+    hipblas_init_matrix(hA, arg, hipblas_client_never_set_nan, hipblas_symmetric_matrix, true);
     hipblas_init_vector(hx, arg, hipblas_client_alpha_sets_nan, false, true);
 
-    hA_cpu.copy_from(hA);
+    // helper function to convert Regular matrix `hA` to packed matrix `hAp`
+    regular_to_packed(uplo == HIPBLAS_FILL_MODE_UPPER, hA, hAp, N);
 
-    CHECK_HIP_ERROR(dA.transfer_from(hA));
+    hAp_cpu.copy_from(hAp);
+
+    CHECK_HIP_ERROR(dAp.transfer_from(hAp));
     CHECK_HIP_ERROR(dx.transfer_from(hx));
-    CHECK_HIP_ERROR(hipMemcpy(d_alpha, &h_alpha, sizeof(T), hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(d_alpha, halpha, sizeof(T), hipMemcpyHostToDevice));
 
     if(arg.unit_check || arg.norm_check)
     {
         /* =====================================================================
             HIPBLAS
         =================================================================== */
         CHECK_HIPBLAS_ERROR(hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_HOST));
-        DAPI_CHECK(
-            hipblasSprBatchedFn,
-            (handle, uplo, N, &h_alpha, dx.ptr_on_device(), incx, dA.ptr_on_device(), batch_count));
-
-        CHECK_HIP_ERROR(hA_host.transfer_from(dA));
-        CHECK_HIP_ERROR(dA.transfer_from(hA));
+        DAPI_CHECK(hipblasSprBatchedFn,
+                   (handle,
+                    uplo,
+                    N,
+                    &h_alpha,
```

Comment:
This just halpha?

Reply:
`h_alpha` is not a vector it contains the value of alpha. But `halpha` is a vector.